setup.sh

```diff
@@ -26,343 +36,288 @@ WARN="${YELLOW}WARNING:$RESET"
+	# enable IP forwarding for docker
+	if [ "$(sysctl -n net.ipv4.ip_forward)" != "1" ]; then
+		if [ -d /etc/sysctl.d ]; then
+			echo "net.ipv4.ip_forward=1" >/etc/sysctl.d/99-tinkerbell.conf
+		elif [ -f /etc/sysctl.conf ]; then
+			echo "net.ipv4.ip_forward=1" >>/etc/sysctl.conf
+		fi
 
-generate_envrc() {
-	# backup existing environment config if any
-	if [ -f "$ENV_FILE" ]; then
-		echo "$INFO found existing $ENV_FILE, moving it to $ENV_FILE.bak"
-		mv "$ENV_FILE" "$ENV_FILE".bak
+		sysctl net.ipv4.ip_forward=1
+	fi
+)
+
+setup_networking_netplan() (
+	jq -n \
+		--arg interface "$TINKERBELL_NETWORK_INTERFACE" \
+		--arg cidr "$TINKERBELL_NETWORK_CIDR" \
+		--arg host_ip "$TINKERBELL_HOST_IP" \
+		--arg nginx_ip "$TINKERBELL_NGINX_IP" \
+		'{
+  network: "hi",
+  renderer: "networkd",
+  ethernets: {
+    ($interface): {
+      addresses: [
+        "\($host_ip)/\($cidr)",
+        "\($nginx_ip)/\($cidr)"
+      ]
+    }
+  }
+}' >"/etc/netplan/${TINKERBELL_NETWORK_INTERFACE}.yaml"
+
+	ip link set "$TINKERBELL_NETWORK_INTERFACE" nomaster
+	netplan apply
+	echo "$INFO waiting for the network configuration to be applied by systemd-networkd"
+	sleep 3
+)
+
+setup_networking_ubuntu_legacy() (
+	if [ ! -f /etc/network/interfaces ]; then
+		echo "$ERR file /etc/network/interfaces not found"
+		exit 1
 	fi
 
-	list_network_interfaces
-	tink_interface=$(get_tinkerbell_network_interface)
-	get_tinkerbell_ips
-	get_registry_credentials
+	if grep -q "$TINKERBELL_NETWORK_INTERFACE" /etc/network/interfaces; then
+		echo "$ERR /etc/network/interfaces already has an entry for $TINKERBELL_NETWORK_INTERFACE."
+		echo "$BLANK To prevent breaking your network, please edit /etc/network/interfaces"
+		echo "$BLANK and configure $TINKERBELL_NETWORK_INTERFACE as follows:"
+		generate_iface_config
+		echo ""
+		echo "$BLANK Then run the following commands:"
+		echo "$BLANK ip link set $TINKERBELL_NETWORK_INTERFACE nomaster"
+		echo "$BLANK ifdown $TINKERBELL_NETWORK_INTERFACE:0"
+		echo "$BLANK ifdown $TINKERBELL_NETWORK_INTERFACE:1"
+		echo "$BLANK ifup $TINKERBELL_NETWORK_INTERFACE:0"
+		echo "$BLANK ifup $TINKERBELL_NETWORK_INTERFACE:1"
+		exit 1
+	else
+		generate_iface_config >>/etc/network/interfaces
+		ip link set "$TINKERBELL_NETWORK_INTERFACE" nomaster
+		ifdown "$TINKERBELL_NETWORK_INTERFACE:0"
+		ifdown "$TINKERBELL_NETWORK_INTERFACE:1"
+		ifup "$TINKERBELL_NETWORK_INTERFACE:0"
+		ifup "$TINKERBELL_NETWORK_INTERFACE:1"
+	fi
+)
 
-	# the following envs will eventually goaway but are required for now
-	echo -e "\nexport FACILITY=onprem" >>"$ENV_FILE"
-	echo -e "export ROLLBAR_TOKEN=ignored" >>"$ENV_FILE"
-	echo -e "export ROLLBAR_DISABLE=1\n" >>"$ENV_FILE"
-}
+generate_iface_config() (
+	cat <<EOF
 
-command_exists() {
-	command -v "$@" >/dev/null 2>&1
-}
+auto $TINKERBELL_NETWORK_INTERFACE:0
+iface $TINKERBELL_NETWORK_INTERFACE:0 inet static
+    address $TINKERBELL_HOST_IP
+    netmask $TINKERBELL_NETMASK
+    broadcast $TINKERBELL_BROADCAST_IP
```

Comment:
we can use just the CIDR address and the scripts will figure out netmask / broadcast on its own

Reply:
Really nice, sorted!